v2/internal/plugins/collector/proxy/manager.go

```diff
@@ -373,7 +369,7 @@ func (cm *ContextManager) DefineMetric(ns string, unit string, isDefault bool, d
 
 	err := cm.metricsDefinition.AddRule(ns)
 	if err != nil {
-		logF.WithFields(moduleFields).WithError(err).WithFields(logrus.Fields{"namespace": ns}).Errorf("Wrong metric definition")
+		logF.WithFields(moduleFields).WithError(err).WithFields(logrus.Fields{"namespace": ns}).Fatal("Wrong metric definition")
```

Comment:
do we really want the plugin to crash in this situation? 
wouldn't this cause some of the inprocess plugins crashing?

Reply:
well, that was your proposal in https://swicloud.atlassian.net/browse/AO-19619.
generally, a mistake in DefineMetric is a developer mistake, not the runtime one. On the other hand I wonder what would happened in case of dynamic DefineMetric based on other tools. We may discuss it during standup